### PR TITLE
ADM-880: [frontend] feat: generate chart - board metrics

### PR DESCRIPTION
**Pull request by neomgb**
## Summary

generate chart - board metrics

## Before

_Description_

**Screenshots**
If applicable, add screenshots to help explain behavior of your code.

## After

_Description_

**Screenshots**
If applicable, add screenshots to help explain behavior of your code.

## Note

_Null_

### CODE REVIEW

**Review by Leiqiuhong — APPROVED** [review]

File: frontend/__tests__/containers/ReportStep/ReportStep.test.tsx [hunk 1]
```diff
@@ -770,5 +784,45 @@ describe('Report Step', () => {
       expect(screen.getByText('Dev Change Failure Rate')).toBeInTheDocument();
       expect(screen.getByText('Dev Mean Time To Recovery')).toBeInTheDocument();
     });
+
+    it('should select DORA tab when click DORA tab from chart page again', async () => {
+      setup(REQUIRED_DATA_LIST, [fullValueDateRange, emptyValueDateRange]);
+
+      const switchChartButton = screen.getByText('Chart');
```
Comment (guzhongren) [inline frontend/__tests__/containers/ReportStep/ReportStep.test.tsx:791]: extract `'Chart'` DORA, Lis ... into a fixture

File: frontend/__tests__/utils/Util.test.tsx [hunk 2]
```diff
@@ -605,3 +606,11 @@ describe('updateResponseCrews function', () => {
     expect(result).toEqual(expectData);
   });
 });
+
+describe('xAxisLabelDateFormatter function', () => {
+  it('should show the correct date format', () => {
+    const inputDate = '2024/01/15-2024/01/19';
+    const result = xAxisLabelDateFormatter(inputDate);
```
Comment (guzhongren) [inline frontend/__tests__/utils/Util.test.tsx:613]: new line betweeen `given when then`

File: frontend/src/containers/ReportStep/BoardMetricsChart/index.tsx [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+      reworkChart.dispose();
+    };
+  }, [rework, reworkData]);
+
+  return (
+    <ChartContainer>
+      <ChartWrapper ref={velocity}></ChartWrapper>
+      <ChartWrapper ref={averageCycleTime}></ChartWrapper>
+      <ChartWrapper ref={cycleTime}></ChartWrapper>
+      <ChartWrapper ref={rework}></ChartWrapper>
```
Comment (guzhongren) [inline frontend/src/containers/ReportStep/BoardMetricsChart/index.tsx:304]: split `chart`s into different components

File: frontend/src/containers/ReportStep/BoardMetricsChart/index.tsx [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+  }, [cycleTime, cycleTimeData]);
+
+  useEffect(() => {
+    const reworkChart = echarts.init(rework.current);
+    const option = reworkData && stackedAreaOptionMapper(reworkData);
+    reworkChart.setOption(option);
+    return () => {
+      reworkChart.dispose();
+    };
+  }, [rework, reworkData]);
```
Comment (guzhongren) [inline frontend/src/containers/ReportStep/BoardMetricsChart/index.tsx:297]: split

File: frontend/src/utils/util.ts [hunk 5]
```diff
@@ -197,3 +197,11 @@ export function combineBoardInfo(boardInfoResponses: BoardInfoResponse[]) {
     };
   }
 }
+
+export const xAxisLabelDateFormatter = (dateRange: string) => {
+  const [startDate, endDate] = dateRange.split('-');
+  const startMonthDay = startDate.slice(5);
```
Comment (guzhongren) [inline frontend/src/utils/util.ts:203]: do we need to check the length of dateRange, startDate, endDate?

**Review by weiraneve — APPROVED** [review]
LGTM

File: frontend/src/utils/util.ts (hunk 5 above)
Comment (neomgb) [reply frontend/src/utils/util.ts:203]: No need, the format of time must be similar to 2024/03-05/2024/03/15.The length of time is fixed.

File: frontend/src/containers/ReportStep/BoardMetricsChart/index.tsx (hunk 3 above)
Comment (neomgb) [reply frontend/src/containers/ReportStep/BoardMetricsChart/index.tsx:304]: 抽出来的话，目录下多四个文件夹感觉又有点冗余。Dora 那块也还没有抽，有必要抽的话再单独提个 pr 一块搞一下吧

**Review by mrcuriosity-tw — COMMENTED** [review]
image
Is this confirmed with @Yufan0226 ?

**Comment by neomgb** [comment]
> image Is this confirmed with @Yufan0226 ?

This is a bug just introduced, I will fix. thx.

**Pull request merged**